rcl/test/rcl/test_wait.cpp

```diff
@@ -178,7 +179,9 @@ TEST(CLASSNAME(WaitSetTestFixture, RMW_IMPLEMENTATION), multi_wait_set_threaded)
             ss << "[thread " << test_set.thread_id << "] Waiting (try #" << wake_try_count << ")";
             // printf("%s\n", ss.str().c_str());
           }
+          std::this_thread::yield();
           ret = rcl_wait(&test_set.wait_set, wait_period);
+          ASSERT_NE(ret, RCL_RET_ERROR);
```

Comment:
I'd move this into the else section of the `if (ret != RCL_RET_TIMEOUT)` and change it to `ASSERT_EQ(ret, RCL_RET_OK)`.